### DISCUSSION TITLE
pip install tts

### DISCUSSION THREAD

**Discussion by Nihaarika13** [coqui-ai/TTS#4025] [General Q&A]
I am facing this error while installing tts through pip:
copying TTS\vocoder\models\wavernn.py -> temp_build\TTS\vocoder\models
      creating temp_build\TTS\vocoder\utils
      copying TTS\vocoder\utils\__init__.py -> temp_build\TTS\vocoder\utils
      copying TTS\vocoder\utils\distribution.py -> temp_build\TTS\vocoder\utils
      copying TTS\vocoder\utils\generic_utils.py -> temp_build\TTS\vocoder\utilsrunning build_ext
      building 'TTS.tts.utils.monotonic_align.core' extension
      creating build\temp.win-amd64-cpython-310\Release\TTS\tts\utils\monotonic_align
      "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.36.32532\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\nihaj\AppData\Local\Temp\pip-build-env-9jb3mvz8\overlay\Lib\site-packages\numpy\_core\include -IC:\Users\nihaj\AppData\Local\Programs\Python\Python310\include -IC:\Users\nihaj\AppData\Local\Programs\Python\Python310\Include "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.36.32532\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\VS\include" /TcTTS/tts/utils/monotonic_align/core.c /Fobuild\temp.win-amd64-cpython-310\Release\TTS/tts/utils/monotonic_align/core.obj
      core.c
      C:\Users\nihaj\AppData\Local\Programs\Python\Python310\include\pyconfig.h(59): fatal error C1083: Cannot open include file: 'io.h': No such file or directory
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.36.32532\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for tts
Failed to build tts
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (tts)

I am facing this problem since many days, though i've installed visual studio installer and installed the required tools and components. The whole output is here:  
[Python error.txt](https://github.com/user-attachments/files/17358884/Python.error.txt)

Python:3.10.11
pip:24.2
windows 11 pc

**Comment by eginhard** [comment]
You can install [our fork](https://github.com/idiap/coqui-ai-TTS) instead with `pip install coqui-tts`, which should work fine on Windows. This repo is not maintained anymore.

**Reply by tuckery** [reply]
What is the difference between this product and xtts?

**Reply by eginhard** [reply]
You can use that fork in exactly the same way as this repository. It mostly contains a lot of bug fixes and is tested to work with recent versions of Python and dependencies. You can see a detailed list of changes here: https://github.com/idiap/coqui-ai-TTS/releases

XTTS is one specific model that you can use with the Coqui TTS toolkit (both in the original repo and the fork).
